<launch>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="model" default="$(find hand_simulator)/urdf/hand_T42.xacro"/> <!-- Change the urdf here to choose the required type of gripper -->
  <!-- <arg name="world" default="$(find hand_simulator)/worlds/hand.world"/> -->

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <!-- Load the URDF into the ROS Parameter Server -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />

  <rosparam command="load" file="$(find hand_simulator)/config/gazebo_position_control.yaml" />

  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
        args="-z 0.0 -unpause -urdf -model hand -param robot_description" respawn="false" output="screen" />

  <node name="spawn_table" pkg="gazebo_ros" type="spawn_model" args="-file $(find hand_simulator)/urdf/table.urdf -urdf -x 0 -z 0 -model table" />

  <node name="spawn_object" pkg="gazebo_ros" type="spawn_model" args="-file $(find hand_simulator)/urdf/object.urdf -urdf -x 0.19 -z 0.6 -model object" />

  <!-- Load custom joint_state_publisher and object_state_publisher -->
  <node pkg="hand_simulator" type="my_joint_state_publisher.py" name="my_joint_state_publisher" output="screen">
    <rosparam command="load" file="$(find hand_simulator)/param/gripper.yaml" />  
  </node>
  <node pkg="hand_simulator" type="object_state_publisher.py" name="object_state_publisher" output="screen"></node>

</launch>

